src/plugins/point_cloud/PointCloud_TEST.cc

```diff
@@ -18,35 +18,156 @@
 };
 
+using namespace std::chrono_literals;
+
 using namespace gz;
 using namespace gui;
 
+class PointCloudTestFixture : public ::testing::Test
+{
+  public:
+    transport::Node node;
+    rendering::ScenePtr scene;
+    transport::Node::Publisher pointcloudPub;
+    transport::Node::Publisher flatPub;
+    msgs::PointCloudPacked pcMsg;
+    msgs::Float_V flatMsg;
+    std::atomic<bool> receivedMsg;
+
+    PointCloudTestFixture()
+    {
+      this->pointcloudPub = this->node.Advertise<msgs::PointCloudPacked>(
+        "/point_cloud");
+      this->flatPub = this->node.Advertise<msgs::Float_V>("/flat");
+
+      this->InitMockData();
+
+      this->node.Advertise("/marker",
+        &PointCloudTestFixture::OnMarkerMsg, this);
+      this->receivedMsg = false;
+    }
+
+    public: void InitMockData()
+    {
+      msgs::InitPointCloudPacked(this->pcMsg, "some_frame", true,
+          {{"xyz", msgs::PointCloudPacked::Field::FLOAT32}});
+
+      int numberOfPoints{1000};
+      unsigned int dataSize{numberOfPoints * this->pcMsg.point_step()};
+      this->pcMsg.mutable_data()->resize(dataSize);
+      this->pcMsg.set_height(1);
+      this->pcMsg.set_width(1000);
+
+      // Populate messages
+      msgs::PointCloudPackedIterator<float> xIter(this->pcMsg, "x");
+      msgs::PointCloudPackedIterator<float> yIter(this->pcMsg, "y");
+      msgs::PointCloudPackedIterator<float> zIter(this->pcMsg, "z");
+
+      for (float x = 0.0, y = 0.0, z = 0.0;
+          xIter != xIter.End();
+          ++xIter, ++yIter, ++zIter)
+      {
+        *xIter = x;
+        *yIter = y;
+        *zIter = z;
+        flatMsg.add_data(x);
+
+        x += 1.0;
+        if (x > 9)
+        {
+          x = 0.0;
+          y += 1.0;
+        }
+        if (y > 9)
+        {
+          y = 0.0;
+          z += 1.0;
+        }
+      }
+    }
+
+    public: void Publish()
+    {
+      this->pointcloudPub.Publish(this->pcMsg);
+      this->flatPub.Publish(this->flatMsg);
+    }
+
+    /// \brief Callback that receives marker messages.
+    /// \param[in] _req The marker message.
+    public: void OnMarkerMsg(const msgs::Marker &_req)
+    {
+      if (_req.action() == msgs::Marker::ADD_MODIFY)
+      {
+        EXPECT_EQ(_req.id(), 1);
+        EXPECT_EQ(_req.ns(), "/point_cloud/flat");
+        EXPECT_EQ(_req.type(), msgs::Marker::POINTS);
+        EXPECT_EQ(_req.visibility(), msgs::Marker::GUI);
+        if (_req.point().size() != 0)
+        {
+          // We might receive empty packets as the sending process
+          // is asynchronuous
+          EXPECT_EQ(_req.point().size(), this->flatMsg.data().size());
+          EXPECT_EQ(_req.materials().size(), this->flatMsg.data().size());
+          this->receivedMsg = true;
+        }
+      }
+      else if(_req.action() == gz::msgs::Marker::DELETE_ALL)
+      {
+        // Do nothing. Its ok to clear the screen.
+      }
+      else
+      {
+        ADD_FAILURE();
+      }
```

Comment:
```suggestion
      else if (_req.action() != msgs::Marker::DELETE_ALL)
      {
        // if DELETE_ALL, its ok to clear the screen. Otherwise fail
        ADD_FAILURE();
      }
```